lib/pbench/server/api/__init__.py

```diff
@@ -213,47 +217,48 @@ def post(self):
                 url = f"{elasticsearch}/{json_data['indices']}"
 
             if "payload" in json_data:
-                response = requests.post(url, json=json_data["payload"])
+                es_response = requests.post(url, json=json_data["payload"])
             else:
-                app.logger.info(
+                app.logger.debug(
                     "No payload found in Elasticsearch post request json data"
                 )
-                response = requests.get(url)
-
-            # construct response object
-            response = make_response(response.text)
-            response = _corsify_actual_response(response)
-
+                es_response = requests.get(url)
         except requests.exceptions.ConnectionError:
             app.logger.exception(
                 "Connection refused during the Elasticsearch post request"
             )
             abort(
-                500, message="Network problem, could not post to Elasticsearch Endpoint"
+                503, message="Network problem, could not post to Elasticsearch Endpoint"
             )
         except requests.exceptions.Timeout:
             app.logger.exception(
                 "Connection timed out during the Elasticsearch post request"
             )
             abort(
-                500,
+                504,
                 message="Connection timed out, could not post to Elasticsearch Endpoint",
```

Comment:
Same...

Reply:
But `504` is `Gateway Timeout`, which is what happened here.  The API server is the "gateway", and it timed out trying to talk to the configured Elasticsearch instance.